### PR TITLE
json import select filter

### PR DESCRIPTION
**Pull request by FinchPowers**
2 nouveautés
* select
* named

### CODE REVIEW

File: plugins/json_importer.cc [hunk 1]
```diff
@@ -110,6 +123,57 @@ struct JsonScope : SqlExpressionMldbScope {
...
 
+    GetAllColumnsOutput
+    doGetAllColumns(const Utf8String & tableName,
+                    std::function<ColumnName (const ColumnName &)> keep) override
+    {
+        std::vector<KnownColumn> columnsWithInfo;
+
+        auto exec = [=] (const SqlRowScope & scope, const VariableFilter & filter)
+        {
+            StructValue result;
```
Comment (FinchPowers) [inline plugins/json_importer.cc]: result.reserve(col size)

File: plugins/json_importer.cc [hunk 2]
```diff
@@ -227,7 +298,7 @@ struct JSONImporter: public Procedure {
         {
             auto & threadAccum = accum.get();
 
-            int64_t actualLineNum = lineNumber + lineOffset;
```
Comment (jeremybarnes) [inline plugins/json_importer.cc]: int64_t is better here.  It is possible that a 32 bit machine may have more than 4 billion lines.

File: plugins/json_importer.cc [hunk 3]
```diff
@@ -251,22 +322,40 @@ struct JSONImporter: public Procedure {
...
-                    return true;
-                }
-            }
-
             skipJsonWhitespace(*parser.context);
             if (!parser.context->eof()) {
                 return handleError("extra characters at end of line", actualLineNum, "");
             }
 
+            unique_ptr<RowName> rowName;
```
Comment (jeremybarnes) [inline plugins/json_importer.cc]: Don't use a unique_ptr here.  It adds a useless alloc.  Use a rowName and assign it directly.

**Comment by jeremybarnes** [comment]
More comments, +1 once addressed.

**Pull request merged**